### DISCUSSION TITLE
How to display animation in trame?

### DISCUSSION THREAD

**Discussion by Theodliu** [Kitware/trame#363] [Q&A] [answered]
Hi, I have a question about the animation. I refer to this https://discourse.vtk.org/t/how-to-play-the-animation-of-gltf-model/3939 but I apply it in trame.
Here's my code:
`class vtkTimerCallback():
    def __init__(self, steps, reader, interactor):
        self.timer_count = 0
        self.steps = steps
        self.reader = reader
        self.interactor = interactor
        self.timerId = None

    def execute(self, obj, event):
        step = 0
        while step < self.steps:
            self.reader.UpdateTimeStep(self.timer_count / 100.0)
            self.reader.Update()
            self.interactor = obj
            self.interactor.GetRenderWindow().Render()
            self.timer_count += 1
            step += 1
        if self.timerId:
            interactor.DestroyTimer(self.timerId)
`
And here is my main function:
`    cb = vtkTimerCallback(100, reader, interactor)
    interactor.AddObserver('TimerEvent', cb.execute)
    cb.timerId = interactor.CreateRepeatingTimer(500)
    reader.EnableAnimation(0)`
   
However, when I start a server, the interactor doesn't start so there is no TimerEvent. Also, even if I start interactor before the server. It won't render the view before the iteration in the execute() finished. I tried this: https://examples.vtk.org/site/Python/Utilities/Animation/

**Comment by jourdain** [comment, marked as answer]
You should use an async loop in trame. What matter is just calling `self.reader.UpdateTimeStep(...)` and update the view.

**Reply by Theodliu** [reply]
It works now. Thanks a lot.
